src/channel_internal.c

```diff
@@ -392,10 +481,10 @@ IMPLEMENT_MOCKABLE_FUNCTION(, CHANNEL_RESULT, channel_internal_push, THANDLE(CHA
         /*Codes_SRS_CHANNEL_INTERNAL_43_124: [ If the first operation in the list of pending operations contains a non-NULL pull_callback: ]*/
         else if (CONTAINING_RECORD(channel_internal_ptr->op_list.Flink, CHANNEL_OP, anchor)->pull_callback != NULL)
```

Comment:
>pull_callback [](http://example.com/codeflow?start=93&length=13)

The fact that this "pull_callback" is evaluated once here and again in `dequeue_operation` in line 309 points to a design flaw.

Why does it need to be evaluated twice?

Reply:
Probably the same for `push_callback`.